### PR TITLE
Added multi-stage docker build

### PR DESCRIPTION
**Pull request by cloudbehl**
- Multi-stage docker build will improve the container pull time

Issue: https://github.com/gluster/gluster-mixins/issues/39

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,46 +1,69 @@
...
 RUN ./build.sh example.jsonnet
 
-#shell script to apply the k8s object to underlying cluster
+# Copy generated files to /
+RUN cp -r manifests /
+
+
+#-- Final container
+
+FROM docker.io/centos:7.5.1804 as final
```
Comment (JohnStrunk) [inline Dockerfile]: Right below this `FROM` line, you're going to want to update the installed packages or quay will show a bunch of open vulnerabilities in the built container. Anthill example: https://github.com/gluster/anthill/blob/eb675e1d6b48b5caa7818d6b845f48e4d51c4e5f/build/Dockerfile#L2

Comment (cloudbehl) [reply Dockerfile]: Ack, Updated the PR.

**Comment by cloudbehl** [comment]
The thing after the update I saw was the size of the image got increased by almost 170MB.

Is there an image that provides the fix and that we can directly pull fro docker or quay?

```
gluster/gluster-mixins            latest              f30e9b470890        About an hour ago    239MB
gluster/gluster-mixins            latest              8b6ba0bc58ed       About an hour ago    408MB

```

**Comment by JohnStrunk** [comment]
You can specify `centos:7` as your base, which may/may not be better. It advances w/ the latest built container in the 7.x series. That's as good as you'll get though.

**Comment by cloudbehl** [comment]
> You can specify `centos:7` as your base, which may/may not be better. It advances w/ the latest built container in the 7.x series. That's as good as you'll get though.

Updated to centos:7 the size is back to 264MB. 

`gluster/gluster-mixins               latest              ae94d169f483        21 minutes ago      264MB`

**Pull request merged**